src/IStepExecutor.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright 2018 ThoughtWorks, Inc.
+//
+// This file is part of Gauge-CSharp.
+//
+// Gauge-CSharp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Gauge-CSharp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Gauge-CSharp.  If not, see <http://www.gnu.org/licenses/>.
+
+using Gauge.Dotnet.Models;
+
+namespace Gauge.Dotnet
+{
+    public interface IStepExecutor
+    {
+        ExecutionResult ExecuteStep(GaugeMethod gaugeMethod, string[] args);
```

Comment:
We don't need `IStepExecutor` and `IHookExecutor` as separate interfaces. The implementation are different types `StepExecutor` and `HookExecutor`, both can have `Execute` method. The interface can be generic.